File: Gaming_Platform/Models/games.py
from Database.db_connection import DatabaseConnection

class Games:

    @staticmethod
    def create_game(id_games,title,genre,price,developer):
        connection=DatabaseConnection.get_connection()
        cursor=connection.cursor()
        try:
            cursor.execute(
                "INSERT INTO games (id_games,title,genre,price,developer) VALUES (%s, %s, %s, %s, %s)",
                (id_games,title,genre,price,developer)
            )
            connection.commit()
            print(f"✅ Game {title} added successfully!")
        except Exception as e:
            print(f"❌ Error adding game: {e}")
            connection.rollback()
        finally:
            cursor.close()
            DatabaseConnection.return_connection(connection)


    @staticmethod
    def get_all_games():
        connection = DatabaseConnection.get_connection()
        cursor = connection.cursor()
        try:
            cursor.execute("SELECT * FROM games")
            games = cursor.fetchall()
            return games
        except Exception as e:
            print(f"❌ Error reading games: {e}")
            return []
        finally:
            cursor.close()
            DatabaseConnection.return_connection(connection)


    @staticmethod
    def update_game_price(id_game, new_price):
        connection = DatabaseConnection.get_connection()
        cursor = connection.cursor()
        try:
            cursor.execute(
                "UPDATE games SET price = %s WHERE id_games = %s",
                (new_price, id_game)
            )
            connection.commit()
            print(f"✅ New price {id_game}!")
        except Exception as e:
            print(f"❌ Error updating price: {e}")
            connection.rollback()
        finally:
            cursor.close()
            DatabaseConnection.return_connection(connection)

    @staticmethod
    def delete_game(id_game):
        connection = DatabaseConnection.get_connection()
        cursor = connection.cursor()
        try:
            cursor.execute("DELETE FROM games WHERE id_games = %s", (id_game,))
            connection.commit()
            print(f"✅ Game with id  {id_game} was deleted!")
        except Exception as e:
            print(f"❌Error deleting the game: {e}")
            connection.rollback()
        finally:
            cursor.close()
            DatabaseConnection.return_connection(connection)

File: Gaming_platform/src/Models/Games.java
package Models;

import DatabaseConnection.DataConnector;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

public class Games {

    public static void createGame(int idGames, String title, String genre, double price, String developer, String dbType) {
        String query = "INSERT INTO games (id_games, title, genre, price, developer) VALUES (?, ?, ?, ?, ?)";
        try (Connection connection = DataConnector.connect(dbType);
             PreparedStatement stmt = connection.prepareStatement(query)) {

            stmt.setInt(1, idGames);
            stmt.setString(2, title);
            stmt.setString(3, genre);
            stmt.setDouble(4, price);
            stmt.setString(5, developer);

            stmt.executeUpdate();
            System.out.println("✅ Game " + title + " added successfully!");

        } catch (SQLException e) {
            System.out.println("❌ Error adding game: " + e.getMessage());
        }
    }

    public static void getAllGames(String dbType) {
        String query = "SELECT * FROM games";
        try (Connection connection = DataConnector.connect(dbType);
             PreparedStatement stmt = connection.prepareStatement(query);
             ResultSet rs = stmt.executeQuery()) {

            while (rs.next()) {
                int id = rs.getInt("id_games");
                String title = rs.getString("title");
                String genre = rs.getString("genre");
                double price = rs.getDouble("price");
                String developer = rs.getString("developer");

                System.out.println("🎮 " + id + ": " + title + " | " + genre + " | " + price + "$ | Developer: " + developer);
            }

        } catch (SQLException e) {
            System.out.println("❌ Error fetching games: " + e.getMessage());
        }
    }

    public static void updateGamePrice(int idGame, double newPrice, String dbType) {
        String query = "UPDATE games SET price = ? WHERE id_games = ?";
        try (Connection connection = DataConnector.connect(dbType);
             PreparedStatement stmt = connection.prepareStatement(query)) {

            stmt.setDouble(1, newPrice);
            stmt.setInt(2, idGame);

            int rowsAffected = stmt.executeUpdate();
            if (rowsAffected > 0) {
                System.out.println("✅ Price updated for game " + idGame + "!");
            } else {
                System.out.println("⚠️ Game with ID " + idGame + " does not exist.");
            }

        } catch (SQLException e) {
            System.out.println("❌ Error updating price: " + e.getMessage());
        }
    }

    public static void deleteGame(int idGame, String dbType) {
        String query = "DELETE FROM games WHERE id_games = ?";
        try (Connection connection = DataConnector.connect(dbType);
             PreparedStatement stmt = connection.prepareStatement(query)) {

            stmt.setInt(1, idGame);

            int rowsAffected = stmt.executeUpdate();
            if (rowsAffected > 0) {
                System.out.println("✅ Game " + idGame + " deleted!");
            } else {
                System.out.println("⚠️ Game with ID " + idGame + " does not exist.");
            }

        } catch (SQLException e) {
            System.out.println("❌ Error deleting game: " + e.getMessage());
        }
    }
}
